packages/shorebird_cli/lib/src/commands/release_new/android_releaser.dart

```diff
@@ -0,0 +1,213 @@
+      .map(
+        (platform) => AndroidArch.availableAndroidArchs
+            .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+      )
+      .toSet();
+
+  /// Whether to generate an APK in addition to the AAB.
+  late bool generateApk = argResults['android-artifact'] as String == 'apk';
+
+  /// Whether to split the APK per ABI. This is not something we support, but
+  /// we check for this to provide a more helpful error message.
+  late bool splitApk = argResults['split-per-abi'] == true;
+
+  @override
+  Future<void> assertPreconditions() async {
+    try {
+      await shorebirdValidator.validatePreconditions(
+        checkUserIsAuthenticated: true,
+        checkShorebirdInitialized: true,
+        validators: doctor.androidCommandValidators,
+      );
+    } on PreconditionFailedException catch (e) {
+      exit(e.exitCode.code);
+    }
+  }
+
+  @override
+  Future<void> assertArgsAreValid() async {
+    if (generateApk && splitApk) {
+      logger
+        ..err(
+          'Shorebird does not support the split-per-abi option at this time',
+        )
+        ..info(
+          '''
+Split APKs are each given a different release version than what is specified in the pubspec.yaml.
+
+See ${link(uri: Uri.parse('https://github.com/flutter/flutter/issues/39817'))} for more information about this issue.
+Please comment and upvote ${link(uri: Uri.parse('https://github.com/shorebirdtech/shorebird/issues/1141'))} if you would like shorebird to support this.''',
+        );
+      exit(ExitCode.unavailable.code);
+    }
+  }
+
+  @override
+  Future<FileSystemEntity> buildReleaseArtifacts() async {
+    final architectures = (argResults['target-platform'] as List<String>)
+        .map(
+          (platform) => AndroidArch.availableAndroidArchs
+              .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+        )
+        .toSet();
+
+    final flutterVersionString = await shorebirdFlutter.getVersionAndRevision();
+
+    final buildAppBundleProgress = logger
+        .progress('Building app bundle with Flutter $flutterVersionString');
+    try {
+      await artifactBuilder.buildAppBundle(
+        flavor: flavor,
+        target: target,
+        targetPlatforms: architectures,
+      );
+    } on ArtifactBuildException catch (e) {
+      buildAppBundleProgress.fail(e.message);
+      exit(ExitCode.software.code);
+    }
+
+    buildAppBundleProgress.complete();
+
+    if (generateApk) {
+      final buildApkProgress =
+          logger.progress('Building APK with Flutter $flutterVersionString');
+      try {
+        await artifactBuilder.buildApk(
+          flavor: flavor,
+          target: target,
+          targetPlatforms: architectures,
+        );
+      } on ArtifactBuildException catch (e) {
+        buildApkProgress.fail(e.message);
+        exit(ExitCode.software.code);
+      }
+      buildApkProgress.complete();
+    }
+
+    return shorebirdAndroidArtifacts.findAab(
```

Comment:
You shouldn't need this call here, `artifactBuilder.buildAppBundle` runs this validation

Reply:
Fixed